File: Week-1/Data structures and Algorithms/Exercise-1/Logger.java
public class Logger {
    
    private static Logger instance;

    private String user;
    private String pass;
 
    private Logger() {}
  
    public static Logger getLogger() {
        if (instance == null) {
            instance = new Logger();
        }
        return instance;
    }

    public void setCredentials(String username, String password) {
        this.user = username;
        this.pass = password;
    }
   
    public void printCredentials() {
        System.out.println("User: " + user);
        System.out.println("Pass: " + pass);
    }
}

File: Week-1/Design principles and  Patterns/Exercise-1/Logger.cs
using System;

public class Logger
{
    private static Logger instance;

    private string user;
    private string pass;

    private Logger() { }
 
    public static Logger GetLogger()
    {
        if (instance == null)
        {
            instance = new Logger();
        }
        return instance;
    }

    public void SetCredentials(string username, string password)
    {
        user = username;
        pass = password;
    }

    public void PrintCredentials()
    {
        Console.WriteLine("User: " + user);
        Console.WriteLine("Pass: " + pass);
    }
}
